js/components/accessibilityControls.js

```diff
@@ -147,11 +145,30 @@ AccessibilityControls.prototype = {
     }
     this.vrRotationAllowed = !isKeyDown; //prevent repeat of keyDown
     this.controller.moveVrToDirection(false, keyDirectionMap[charCode]); //stop rotation if isKeyDown === false or prevent prev rotation if press a button (isKeyDown === true)
+
+    var inPrevKeyPressedArrIndex = -1;
+    //check if button code is already in list of pressed buttons (this.prevKeyPressedArr)
+    //if code is in the array return index of the code
+    for (var i = this.prevKeyPressedArr.length - 1; i >= 0; i--) {
+      if (this.prevKeyPressedArr[i] === charCode) {
+        inPrevKeyPressedArrIndex = i;
+        break;
+      }
+    }
     if (isKeyDown === true) {
-      this.controller.moveVrToDirection(true, keyDirectionMap[charCode]);
-      return true;
+      this.prevKeyPressedArr.push(charCode);
+    } else { // if button is up remove a code of the button from this.prevKeyPressedArr
+      if (inPrevKeyPressedArrIndex > -1) {
+        this.prevKeyPressedArr.splice(inPrevKeyPressedArrIndex, 1);
+      }
+    }
+    if (this.prevKeyPressedArr.length) {
+      isKeyDown = true;
+      charCode = this.prevKeyPressedArr[this.prevKeyPressedArr.length-1];
     }
-    return false;
+    //rotate if a button is pressed, stop rotate if other case
+    this.controller.moveVrToDirection(isKeyDown, keyDirectionMap[charCode]);
```

Comment:
So we are only rotating based on what the last key pressed is?  You can't move in two directions at once? Otherwise why keep an array of what is pressed. You could just keep track of what the last key pressed is.

Reply:
Yes, but situations when a user pressed several buttons in the one time are happened, and when the user unpressed each of the buttons we need to keep rotation in current direction. For example, an order of pressing is D & A & W, so firstly we get right rotation, after that we get left rotation, an in the end we get top rotation. If the user unpressed A and after that unpressed W, so we need to rotate to right, but right rotation is not last. So we need to save last 0-4 rotations for the situation.

Reply:
Sounds good. Thank you for the explanation.